### ISSUE TITLE
Perform the tests against IEnumerable, ICollection, IList, ...

### ISSUE THREAD

**Issue by Orace** [morelinq/MoreLINQ#676]
Some methods like SkipLast:

https://github.com/morelinq/MoreLINQ/blob/c3516e335230f2a598ea0ab8f570b3d1b45b0970/MoreLinq/SkipLast.cs#L43-L48

Use different strategies if the received `IEnumerable` parameters can be cast to `ICollection` or `IList`.
But the tests may not be done against all those types.
In such case we have to perform the tests with all relevant object type.

First we have to list those methods:

- [ ] SkipLast
- ...

**Comment by Orace** [comment]
This is related to #670

**Comment by atifaziz** [comment]
> In such case we have to perform the tests with all relevant object type.

Yes, I agree. Please go ahead with a PR if you would like to help with this. Thanks!
